bundle/src/main/java/com/adobe/acs/commons/util/compatability/impl/AemCompatibilityImpl.java

```diff
@@ -0,0 +1,76 @@
+package com.adobe.acs.commons.util.compatability.impl;
+
+import com.adobe.acs.commons.util.compatability.AemCompatibility;
+import com.adobe.granite.license.ProductInfo;
+import com.adobe.granite.license.ProductInfoProvider;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.framework.Version;
+import org.osgi.service.component.annotations.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+@Component(service = {})
+public class AemCompatibilityImpl implements AemCompatibility {
+    private static final Logger log = LoggerFactory.getLogger(AemCompatibilityImpl.class);
+    private static final String PN_HOSTING = "hosting";
+    private static final String PN_VERSION = "version";
+
+    // This is the first Major/Minor GA Version of AEM as a Cloud Service
+    private static final Version originalCloudServiceVersion = new Version(2019, 12,   0);
+
+    @Reference(
+            policy = ReferencePolicy.DYNAMIC,
+            policyOption = ReferencePolicyOption.GREEDY
+    )
+    private transient ProductInfoProvider productInfoProvider;
+    private transient ProductInfo productInfo;
+
+    @Override
+    public Hosting getHosting() {
+        if (productInfo.getVersion().compareTo(originalCloudServiceVersion) > 0) {
```

Comment:
is it ok to derive that from the version? What about running with AEM SDK 20xx locally? Shouldn't we rather derive the hosting option from the documented run modes from AEMaaCS (https://docs.adobe.com/content/help/en/experience-manager-cloud-service/implementing/deploying/overview.html#runmodes)?

Reply:
so - i guess there's some ambiguity here...  here's how i think of it:

* AEM as a Cloud Service (Cloud)
   * Running in the Adobe Cloud 
   * AEM SDK QuickStart Jar running locally
* AMS/On-Prem (Non-Cloud)
   * Any 6.x based AEM deployment

For "AEM as a Cloud Service" .. i'd think we'd want to limit the local AEM SDK QuickStart Jar to run the same feature set as in the "real" Cloud, otherwise, devs might get confused as to why something works locally but not when deployed.

Are you thinking there are 3 types we should target?

* AEM as a Cloud Service running in the Cloud
* AEM as a Cloud Service SDK QuickStart Jar running locally for developers
* AEM 6.x (AMS and OnPrem)

Also - perhaps "hosting" is a bad term ..

Reply:
I think indeed the property should be named differently. Cannot come up with a good proposal right now... Maybe something like "OakCompositeNodeStore" or "StrictMutableImmutableSeparation".

Reply:
@kwin what's the usecase for running something on local dev but not in the cloud?

Reply:
most probably one should rather rely on a specific property whether a mount point is read-only or not.

Reply:
That gets us a split at: "AEM running IN the Cloud"  and "Everything else" ... as noted before i see the split at the Product level: AEM as a Cloud Service (including local dev quickstart) and AEM 6.x (AMS/OnPrem) ...

We shouldnt be encouraging use of features on AEM as a CS local dev that wont work when deployed to AEM in the Cloud. Thats just confusing.

Reply:
then maybe "type" is the right property name here because in fact it is different product types

Reply:
Type, mode...

Also, is there a compelling reason to differentiate between AEM as a cloud service in the cloud and local dev? If there is we should try to figure out what app the splits are now and how to detect. I’m in favor of the 2 types split by logical product (cloud vs 6.x) but if we have a good reason to add further granularity that justifies the additional complexity, we could do it (assuming we can detect any new  splits)